File: src/Data/OrderStatus.java
package Data;

public enum OrderStatus {
	Closed("Closed"),
	Delivered("Delivered"),
	New("New"),
	OnHold("onHold");
	private String name;

	OrderStatus(String name) {
		this.name = name;

	}

	public static OrderStatus findByName(String name) {
		for (OrderStatus o : OrderStatus.values()) {
			if (name.equalsIgnoreCase(o.getName())) {
				return o;
			}
		}
		return OrderStatus.valueOf(OrderStatus.class, name);
	}

	public String getName() {
		return name;
	}

}

File: src/data/OrderStatus.kt
package data

enum class OrderStatus(val Name: String) {
    Closed("Closed"),
    Delivered("Delivered"),
    New("New"),
    OnHold("onHold");


    companion object {

        fun findByName(Name: String): OrderStatus {
            for (o in values()) {
                if (Name.equals(o.Name, ignoreCase = true)) {
                    return o
                }
            }
            return valueOf(Name)
        }

    }
}

